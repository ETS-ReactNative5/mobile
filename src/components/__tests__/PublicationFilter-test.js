import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import PublicationFilter from '../PublicationFilter'

const disciplines = ['science']

it('renders correctly', () => {
  const tree = renderer.create(
    <PublicationFilter isConnected={true} disciplines={disciplines} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render when not connected', () => {
  const tree = renderer.create(
    <PublicationFilter isConnected={false} disciplines={disciplines} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders selected discipline', () => {
  const tree = renderer.create(
    <PublicationFilter isConnected={false} disciplines={disciplines} selectedDiscipline={'science'} />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
