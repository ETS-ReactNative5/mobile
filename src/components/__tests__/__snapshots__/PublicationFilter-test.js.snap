exports[`test does not render when not connected 1`] = `
<View
  style={
    Array [
      undefined,
      false,
    ]
  }>
  <View
    style={undefined}>
    <TouchableOpacity
      activeOpacity={0.7}
      onPress={[Function]}
      style={undefined}>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
            ]
          }>
          Choose Category
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontFamily": "FontAwesome",
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`test renders correctly 1`] = `
<View
  style={
    Array [
      undefined,
      false,
    ]
  }>
  <View
    style={undefined}>
    <TouchableOpacity
      activeOpacity={0.7}
      onPress={[Function]}
      style={undefined}>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
            ]
          }>
          Choose Category
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontFamily": "FontAwesome",
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;

exports[`test renders selected discipline 1`] = `
<View
  style={
    Array [
      undefined,
      false,
    ]
  }>
  <View
    style={undefined}>
    <TouchableOpacity
      activeOpacity={0.7}
      onPress={[Function]}
      style={undefined}>
      <View
        style={undefined}>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              undefined,
            ]
          }>
          Viewing Science
        </Text>
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": undefined,
                "fontFamily": "FontAwesome",
                "fontSize": 12,
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              undefined,
            ]
          }>
          
        </Text>
      </View>
    </TouchableOpacity>
  </View>
</View>
`;
